### PR TITLE
No decorators in storyshots

### PR DESCRIPTION
**Pull request by usulpro**
Issue:

## How to test

Is this testable with jest or storyshots?

Does this need a new example in the kitchen sink apps?

Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by UsulPro** [comment]
cc @coreylight

**Comment by coreylight** [comment]
Thanks @UsulPro!

File: examples/react-native-vanilla/__tests__/__snapshots__/storyshots.js.snap [hunk 1]
```diff
@@ -97,77 +97,68 @@ exports[`Storyshots Button with text 1`] = `
...
-  style={
-    Object {
-      "backgroundColor": "#ffff00",
-      "borderColor": "#ff00ff",
-      "borderWidth": 3,
-      "padding": 10,
-    }
-  }
->
+Array [
```
Comment (ndelangen) [inline examples/react-native-vanilla/__tests__/__snapshots__/storyshots.js.snap]: Looks like this is NOT what we want to happen. Part of the actual story is gone from the snapshot!

**Review by ndelangen — CHANGES_REQUESTED** [review]

**Comment by ndelangen** [comment]
Should we keep a list of all the decorators and only filter out those components?

We should also not just remove the first parent, but we'd have to recursively remove/filter them from top, since decorators could be wrapped (multiple decorators around 1 story).

**Comment by coreylight** [comment]
@ndelangen I like those suggestions. I'm not very familiar with the codebase so I fully expected some iteration on this 😊

**Comment by danielduan** [comment]
I'm gonna close this PR since there hasn't been much activity here. Feel free to reopen or leave a comment here so we can reopen it for you.

**Comment by axelnormand** [comment]
I'm still very keen on not seeing decorators in the storyshot as I use enzyme shallow so the storyshot becomes quite useless :)
Hopefully I can pick up this PR and fix it shortly

**Comment by axelnormand** [comment]
In the meantime my plan is to use ".dive()" in my storyshots.test.ts file to recurse down to find story in question and use that in the call to `toMatchSpecificSnapshot`

**Comment by ggarek** [comment]
Guys, is there any resolution to this issue? @ndelangen @axelnormand 

I am having same issue: **i have my decorators in all the storyshots.**

I was using `"@storybook/addon-storyshots": "^3.4.7"` and it seem not to be an issue.
Then i upgraded to `"@storybook/addon-storyshots": "^4.1.6"` _(current latest)_ and i hit the problem.

**Comment by axelnormand** [comment]
Alas i didn't have time (yet) to contribute anything to the storybook project which might help this.

What i ended up doing was mocking the `addDecorator` method in my jest storyshot tests so it just renders the story (ie does nothing), therefore no decorator appears in the snapshot

```
// mock addDecorator so they are not in snapshot
jest.mock('@storybook/react-native', () => {
  const actual = require.requireActual('@storybook/react-native');
  return {
    ...actual,
    storiesOf: (...args: any[]) => {
      const storybook = actual.storiesOf(...args);
      storybook.addDecorator = () => storybook;
      return storybook;
    },
    addDecorator: () => null,
  };
});
```

**Comment by ndelangen** [comment]
That's an interesting solution @axelnormand.

In 5.0.0 I added new api's for storyshots to be able to **ACCESS** the original storyFn and a fully decorated Fn.

It's currently using the fully decorated Fn always, but it could fairly easily start rendering the raw one based on an option/parameter.

I think that's really the way forward, since some people will WANT the decorators and some DON'T. And quite possibly it might be different per story!

**Comment by ndelangen** [comment]
I might take a crack at it.

**Comment by axelnormand** [comment]
oh didnt know that about the new 5 API, great stuff. Thanks @ndelangen

**Comment by ndelangen** [comment]
@axelnormand we cleanup up A LOT of internals, making us better able to make changes like this. On top of that the `addParameters` api has really given a robust way of configuring anything on a global or story level.

**Comment by ndelangen** [comment]
Gonna open a PR in a few minutes adding this feature!

**Comment by aralroca** [comment]
@axelnormand very great workaround. However, I have some decorators that are adding Context Providers to the stories... Without it, stories don't consume correctly the data... So I need to mock also all these Context...? 😯 Or maybe there is a better way to add these Context Providers than using decorators?

**Pull request closed without merging**